azure/azure_style.yaml

```diff
@@ -132,3 +135,31 @@ jobs:
 
           find . -iname '*.h' -o -iname '*.hpp' -o -iname '*.c' -o -iname '*.cpp' | xargs clang-format --verbose --dry-run --Werror -style=file
 
+  - job: fprettify
+    pool:
+      vmImage: "ubuntu-20.04"
+    timeoutInMinutes: ${{ parameters.TIMEOUT }}
+    condition: ${{ parameters.FPRETTIFY }}
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: "3.9"
+      - script: |
+          # Install fprettify
+          pip install fprettify==0.3.7
+
+          # Run fprettify on non-differentiated Fortran 90 source files
+          find . -type f -iname '*.f90' ! -iname '*_b.f90' ! -iname '*_d.f90' -print0 | while read -d $'\0' file
+          do
+            echo Checking $file
+            fprettify -i=4 -l=120 --whitespace-multdiv=True --strict-indent $file
+
+            # Check if this file changed and print a message if it did
+            git diff --summary --exit-code $file
+            if [ $? -ne 0 ]; then
+              echo $file needs to be formatted
+            fi
+          done
+
+          # Exit with an error if any of the tracked files changed
+          git diff --summary --exit-code
```

Comment:
Is this needed if we already run `git diff` for each file manually? Does the Azure job fail immediately if one file fails, or does it go through and print out all violations before exiting? I think we would prefer the second option if possible.

Reply:
This is needed because the exit code will be 0 if the last file in the loop is properly formatted. When I tried this locally, this printed all violations before exiting. I think it will be the same on Azure.